src/ws/ws_manager.rs

```diff
@@ -358,18 +389,7 @@ impl WsManager {
         subscriptions.remove(index);
 
         if subscriptions.is_empty() {
-            let payload = serde_json::to_string(&SubscriptionSendData {
-                method: "unsubscribe",
-                subscription: &serde_json::from_str::<serde_json::Value>(&identifier)
-                    .map_err(|e| Error::JsonParse(e.to_string()))?,
-            })
-            .map_err(|e| Error::JsonParse(e.to_string()))?;
-
-            let mut writer = self.writer.lock().await;
-            writer
-                .send(protocol::Message::Text(payload))
-                .await
-                .map_err(|e| Error::Websocket(e.to_string()))?;
+            Self::send_subscribe(self.writer.lock().await.borrow_mut(), identifier.as_str()).await?;
```

Comment:
shouldn't this be unsubscribe?

Reply:
Yes, thanks for catching this! Was an error in updating my changes.